src/lib/reader.ts

```diff
@@ -91,39 +87,34 @@ class Reader {
    * Look at a fixed-length unsigned integer, but don't advance the cursor.
    *
    * @param {number} length Length of the integer in bytes.
-   * @return {number} Contents of the next byte.
    */
-  peekUInt (length: number) {
+  peekUInt (length: number): BigNumber {
     if (length === 0) {
-      return 0
-    }
-    if (length < 0) {
+      return new BigNumber(0)
+    } else if (length < 0) {
       throw new Error('Tried to read integer with negative length (provided: ' +
         length + ')')
+    } else if (length > 8) {
+      throw new Error('UInts longer than 8 bytes must be encoded as VarUInts')
     }
-    if (length > Reader.MAX_INT_BYTES) {
-      throw new Error('Tried to read too large integer (requested: ' +
-        length + ', max: ' + Reader.MAX_INT_BYTES + ')')
-    }
-    this.ensureAvailable(length)
-    const value = this.buffer.readUIntBE(this.cursor, length)
-    return value
+
+    return new BigNumber(this.peek(length).toString('hex'), 16)
```

Comment:
This seems slightly janky; couldn't we compute the number manually? Something like:

```js
return this.peek(length).reduce((sum, byte) =>  sum.times(256).plus(byte), new BigNumber(0))
```